Rules/AvoidAssignmentToAutomaticVariable.cs

```diff
@@ -33,6 +33,16 @@ public class AvoidAssignmentToAutomaticVariable : IScriptRule
             "IsCoreCLR", "IsLinux", "IsMacOS", "IsWindows"
         };
 
+        private static readonly IList<string> _automaticVariablesThatCouldBeProblematicToAssignTo = new List<string>()
```

Comment:
I feel we should avoid embedding clausal descriptions in variable names if we can -- and also I can't think of an automatic variable that's not problematic to assign to (since, being problematic, they could be reassigned unexpectedly)

```suggestion
        private static readonly IList<string> _automaticVariables = new List<string>()
```

Reply:
I wanted to differentiate from the `_readOnlyAutomaticVariables` variable above where PowerShell actually itself throws an error when trying to assign, those variables don't.

Reply:
Well one is readonly variables and the other is just automatic variables, no? So `_readonlyVariables` and `_automaticVariables` or `_writeableAutomaticVariables` maybe?

Reply:
I'd just prefer not to call the variable something that embeds a sentence

Reply:
How about `_writeableAutomaticVariables` and `_readonlyAutomaticVariables`?

Reply:
👌